src/bootman/kernel.c

```diff
@@ -233,7 +233,7 @@ Kernel *boot_manager_inspect_kernel(BootManager *self, char *path)
 KernelArray *boot_manager_get_kernels(BootManager *self)
 {
         KernelArray *ret = NULL;
-        DIR *dir = NULL;
+        autofree(DIR) *dir = NULL;
```

Comment:
This is an improvement to existing code which can be easily extracted into a separate commit (along with the `DEF_AUTOFREE` in `bootman.h`).